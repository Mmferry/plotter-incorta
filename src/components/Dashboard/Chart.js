import { useState, useEffect } from "react";
import { Line } from "react-chartjs-2";

const options = {
  scales: {
  }
};

const colorPicker = ["#08457c", "#1e81b0", "#e07b39"]

const Chart = ({ payload }) => {
  const [data, setData] = useState();
  useEffect(() => {
    let datasets = [];
    if (payload) {
      for (let i = 1; i < payload.length; i++) {
        datasets.push({
          label: payload[i].name,
          data: payload[i].values,
          fill: false,
          backgroundColor: colorPicker[i-1],
          borderColor: colorPicker[i-1],
          yAxisID: payload[i].name,
        });
      }
    }

    setData({
      labels: payload[0].values,
      datasets: [...datasets],
    });
  }, [payload]);

  return (
    <div>
      <Line data={data} options={options} />
    </div>
  );
};

export default Chart;
