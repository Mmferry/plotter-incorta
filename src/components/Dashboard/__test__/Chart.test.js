import { mount } from "enzyme";

import DataResponse from "../../../mock/data.mock.json";
import Chart from "../Chart";

describe('<Chart />', () => {
  it('Renders a Line chart', () => {
    const wrapper = mount(
      <Chart payload={DataResponse} />,
    )
    expect(wrapper).toMatchSnapshot()
  });
});