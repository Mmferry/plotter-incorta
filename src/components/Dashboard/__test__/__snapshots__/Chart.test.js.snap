// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> Renders a Line chart 1`] = `ReactWrapper {}`;
